pystac/extensions/version.py

```diff
@@ -47,17 +60,122 @@ class VersionRelType(StringEnum):
     """Indicates a link pointing to a resource containing the successor version in the
     version history."""
 
+    HISTORY = "version-history"
+    """This link points to a version history or changelog.
+    
+    This can be for example a Markdown file with the corresponding media type or
+    a STAC Catalog or Collection.
+    """
+
 
-class VersionExtension(
+class BaseVersionExtension(
     Generic[T],
     PropertiesExtension,
-    ExtensionManagementMixin[Union[pystac.Collection, pystac.Item]],
+    ExtensionManagementMixin[Union[Collection, Item, Catalog]],
+):
+    """A base extension that just gets and sets attributes, not links.
+
+    Used for Assets and AssetDefinitions.
+    """
+
+    name: Literal["version"] = "version"
+
+    def apply_base(
+        self,
+        version: str | None = None,
+        deprecated: bool | None = None,
+        experimental: bool | None = None,
+    ) -> None:
+        """Applies attributes to this extension object."""
+        self.version = version
+        self.deprecated = deprecated
+        self.experimental = experimental
+
+    @property
+    def version(self) -> str | None:
+        """Get or sets a version string of this object."""
+        return self._get_property(VERSION, str)
+
+    @version.setter
+    def version(self, v: str) -> None:
+        self._set_property(VERSION, v, pop_if_none=True)
+
+    @property
+    def deprecated(self) -> bool | None:
+        """Get or sets whether this object is deprecated.
+
+        A value of ``True`` specifies that the object is deprecated with the
+        potential to be removed. It should be transitioned out of usage as soon
+        as possible and users should refrain from using it in new projects. A
+        link with relation type ``latest-version`` SHOULD be added to the links
+        and MUST refer to the resource that can be used instead.
+
+        NOTE:
+            A :class:`pystac.DeprecatedWarning` is issued if the ``deprecated``
+            property is ``True`` when deserializing a dictionary to an object.
+            The :meth:`~pystac.extensions.version.ignore_deprecated` context
+            manager is provided as a convenience to suppress these warnings:
+
+            >>> with ignore_deprecated():
+            ...    deprecated_item = pystac.Item.from_dict(deprecated_item_dict)
+        """
+        return self._get_property(DEPRECATED, bool)
+
+    @deprecated.setter
+    def deprecated(self, v: bool | None) -> None:
+        self._set_property(DEPRECATED, v)
```

Comment:
I think this should get also `pop_if_none` or it should not accept None as an input. 
```suggestion
        self._set_property(DEPRECATED, v, pop_if_none=True)
```